src/assets/scripts/client/airport/AirportWindModel.js

```diff
@@ -0,0 +1,162 @@
+     * @param initialAirportWind {object}  wind value defaults from airport json
+     * @private
+     */
+    _init(initialAirportWind) {
+        if (_isNil(initialAirportWind)) {
+            throw new Error('Invalid wind data provided to AirportWindModel. Expected and object with keys `angle` and `speed`.');
+        }
+
+        this.speed = initialAirportWind.speed;
+        this.angle = initialAirportWind.angle;
+
+        this._eventBus = EventBus;
+
+        return this;
+    }
+
+    /**
+     * Setup handler methods with proper scope binding
+     *
+     * @for AirportWindModel
+     * @method _setupHandlers
+     * @chainable
+     * @private
+     */
+    _setupHandlers() {
+        this._calculateNextWindHandler = this._calculateNextWind.bind(this);
+
+        return this;
+    }
+
+    /**
+     * Enable the instance
+     *
+     * @for AirportWindModel
+     * @method enable
+     * @chainable
+     */
+    enable() {
+        this._createWindUpdateTimer();
+
+        return this;
+    }
+
+    /**
+     * Calculates the current wind vector based on an aircraft's altitude
+     *
+     * @for AirportWindModel
+     * @param altitude {number}
+     * @return windVector {number}
+     */
+    calculateWindVector(altitude) {
+        // FIXME: taken from the `AircraftModel`. this class should provide a method that does this
+        const windIncreaseFactorPerFoot = 0.00002;  // 2.00% per thousand feet
+        const windTravelDirection = this.angle + Math.PI;
+        const windTravelSpeedAtSurface = this.speed;
+        const windTravelSpeed = windTravelSpeedAtSurface * (1 + (altitude * windIncreaseFactorPerFoot));
+        const windVector = vscale(vectorize_2d(windTravelDirection), windTravelSpeed);
+
+        return windVector;
+    }
+
+    /**
+     * Calculate next wind on a bell curve.
+     *
+     * @for AirportWindModel
+     * @method calculateNextWind
+     * @private
+     */
+    _calculateNextWind() {
+        // We need to make sure that none of the properties are 0.
+        // Otherwise, the function will always return 0, because 0 * anything = 0.
+        // If the speed is less than 2, it tends to get stuck on '1' for a while.
+        if (this.speed < 2) {
+            this.speed = 5;
+        }
+
+        const speed = calculateNormalDistributedNumber(this.speed);
+        const angle = calculateNormalDistributedNumber(this.angle + 360);
+
+        this.speed = _clamp(_round(speed), 0, 25);
+        this.angle = degrees_normalize(_round(angle));
+
+        this._eventBus.trigger(EVENT.AIRPORT_WIND_CHANGE);
+    }
+
+    /**
+     * Creates an interval timer that will call `._calculateNextWind()`
+     *
+     * @for AirportWindModel
+     * @method createWindUpdateTimer
+     * @return {gameTimeout} an instance of the new game timeout.
+     * @private
+     */
+    _createWindUpdateTimer() {
+        GameController.game_interval(
+            this._calculateNextWindHandler,
+            300,
```

Comment:
This value should be in milliseconds.  so the 'fires constantly' thing you mentioned is because this value is too small.  As written, this is 3/10s of a second.  I think you may have meant to use `3000` instead, which is 3 seconds.  I'd actually suggest a larger value here, something more like once every 5 minutes or something?

Reply:
Actually, that's after it's been fixed - the wind updates after 5 minutes of in-game time. It appears that the `game_interval` accepts seconds.

Reply:
I double checked and @sam-irl is correct. `game_interval` is using seconds, not milliseconds.